identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp

```diff
@@ -163,13 +163,24 @@
     String multiOptionURIParam = "";
     if (localAuthenticatorNames.size() > 1 || idpAuthenticatorMapping != null && idpAuthenticatorMapping.size() > 1) {
         String baseURL;
-        try {
-            baseURL = ServiceURLBuilder.create().addPath(request.getRequestURI()).build().getRelativePublicURL();
-        } catch (URLBuilderException e) {
-            request.setAttribute(STATUS, AuthenticationEndpointUtil.i18n(resourceBundle, "internal.error.occurred"));
-            request.setAttribute(STATUS_MSG, AuthenticationEndpointUtil.i18n(resourceBundle, "error.when.processing.authentication.request"));
-            request.getRequestDispatcher("error.do").forward(request, response);
-            return;
+        // Check whether authentication endpoint is hosted externally.
+        String isHostedExternally = application.getInitParameter("IsHostedExternally");
+        if (isHostedExternally != null && Boolean.parseBoolean(isHostedExternally)) {
+            String requestURI = request.getRequestURI();
+            if (requestURI != null && StringUtils.isNotBlank(requestURI)) {
```

Comment:
StringUtils.isNotBlank will handle the null check as well

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4104/commits/4ed8f3d89c00be3d76e9bc08f501320f8d1b967e